### ISSUE TITLE
Adding native support on MacOS using Hypervisor Framework.

### ISSUE THREAD

**Issue by ArKam** [hioa-cs/IncludeOS#689]
As Apple now provides a native C APIs to manage VMs in userspace as referenced [here](https://developer.apple.com/library/mac/documentation/DriversKernelHardware/Reference/Hypervisor/), wouldn’t it be nice to have it supported as a native hypervisor instead of having to install VBox?

**Comment by amirmc** [comment]
It’s also worth being aware of projects that make use of Hypervisor.framework.
i.e. https://github.com/docker/hyperkit

**Comment by avsm** [comment]
Yes, we're happy to have contributions to HyperKit that improve its ability to interface with support libraries such as IncludeOS.  There is already some scaffolding in there to link to MirageOS to provide the QCow block layer.  Feel free to raise issues at https://github.com/docker/hyperkit/issues  with any specific questions that arise during IncludeOS integration...

**Comment by fwsGonzo** [comment]
Will take a look at implementation after vacation, and maybe as late as after CppCon. :sunny:

**Comment by ArKam** [comment]
@avsm As hyperKit is specifically develop for Docker, I don’t think IncludeOS should use it or base any development on it as it’ll add another layer on top of something really OS Specific.

In my humble opinion, IncludeOS should keep it on its own and properly integrate Hypervisor framework and not an abstract of it especially when hyperkit is only available on MacOS for now and so do not add a multi-plateform layer that could be a valid argument at integrating it instead of having your own implementation.

@fwsGonzo Wonderful!!! That is just an enhancement issue, not a bug or a critical, but I think it could help developers interested on the “Unikernel” topic to adopt IncludeOS as their preferred framework.

**Comment by alfred-bratterud** [comment]
@ArKam thanks for posting - and @amirmc thanks for the link - I was looking for it earlier :-) I'm currently working on multiboot support for IncludeOS which should make it much easier to boot on various platforms. Looks like hyperkit/xhyve has virtio block and virtio net so that's a big plus.
